hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/WritableUpgradeStore.java

```
@@ -70,7 +72,7 @@ public void resetFileContents() {
 
     @Nullable
     public File removeIf(@NonNull Predicate<File> predicate) {
-        return writableUpgradeFileState.get();
+        return null;
```

Is this intentionally set to null ?

---

yes, removeIf returns null if there is nothing to remove, and this class never removes the file.
Really, I think we might have erred in implementing the WritableQueueState interface.

---

Same question here.

---

@jsync-swirlds  The n we should probably throw `UnsupportedOperationException`

---

It's not unimplemented or unsupported, it's correctly returning that there is nothing left to remove.
That said, what we really need to do is not implement `WritableQueueState` here (I am now convinced that was a mistake), which will remove this method as unnecessary.
@povolev15 Can you make that change?  Remove the `implements WritableQueueState` clause and the overridden methods?  I believe it will not break anything and will simplify this class.

---

sure I will remove and test it